### DISCUSSION TITLE
feature preprocessing/hyperarameters optimization

### DISCUSSION THREAD

**Discussion by Christophe31300** [pycaret/pycaret#1179] [Q&A]
Hi,

First, thank you for providing us a very good autoML package.

I have some questions regarding the exact way some functions are working:
- I wonder how the feature preprocessing (remove multicollinearity, feature selection) is done. Is it done on the whole dataset, on the training set (70%) only, or inside the cross-validation for each training sets independently from the test folds?
- When tuning the hyperparameters of RandomForest for example, I get worse results than with no tuning (using only create_model), even when I try with 200 iterations (I only get this situation once for the moment). So I wanted to know if when tuning models the hyperparameters used by default in the create_model function are also part of the searching space in the tune_model function or not.

Thank you in advance for your answer.

**Comment by Yard1** [comment]
> I wonder how the feature preprocessing (remove multicollinearity, feature selection) is done. Is it done on the whole dataset, on the training set (70%) only, or inside the cross-validation for each training sets independently from the test folds?

Right now the transformers are fitted on the train dataset. We are moving towards the cross validation per fold approach in the next major release.

> When tuning the hyperparameters of RandomForest for example, I get worse results than with no tuning (using only create_model), even when I try with 200 iterations (I only get this situation once for the moment). So I wanted to know if when tuning models the hyperparameters used by default in the create_model function are also part of the searching space in the tune_model function or not.

Most of the times, but not always. Random Forest is one of those exceptions. It is not always possible to account for default parameters in the search space due to different format.

**Reply by Christophe31300** [reply]
Thank you for your answer. Very clear.
You're are doing a great job!
